File: contest/260/a/a.cpp
/*
 * Author: Zedbar
 * Date: 2021-06-21
 */

#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
#include <map>
#include <utility>
#include <math.h>

using namespace std;

typedef long long ll;
typedef pair<int, int> ii;
typedef vector<int> vi;
typedef vector<ii> vii;

int main() {
    unsigned long long int a, b, n;
    cin >> a >> b >> n;
    for (int i = 0; i < n; i++) {
        int tmp = a;
        for (int j = a * 10; j < (tmp + 1) * 10; j++) {
            if (j % b == 0) {
                a = j;
                break;
            }
        }
        if (a == tmp) {
            cout << -1 << endl;
            return 0;
        }
    }
    cout << a << endl;
    return 0;
}

File: contest/260/a/a.java
/*
 * Author: Zedbar
 * Date: 2021-06-21
 */

import java.util.Scanner;
import java.math.BigInteger;

public class a {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int b, n;
        BigInteger a = BigInteger.valueOf(in.nextInt());
        b = in.nextInt();
        n = in.nextInt();

loop: for (int i = 0; i < n; i++) {
          BigInteger tmp = a;
          for (BigInteger j = a.multiply(BigInteger.TEN); j.compareTo(tmp.add(BigInteger.ONE).multiply(BigInteger.TEN)) > 0; j = j.add(BigInteger.ONE)) {
              System.out.println(a.multiply(BigInteger.TEN) + ", " + tmp.add(BigInteger.ONE).multiply(BigInteger.TEN));
              if (j.remainder(BigInteger.valueOf(b)).compareTo(BigInteger.ZERO) == 0) {
                  a = j;
                  continue loop;
              }
              System.out.println(-1);
              return;
          }
}
System.out.println(a);
    }
}
